src/editor.rs

```diff
@@ -187,26 +187,36 @@ impl Editor {
     fn rx(&self) -> usize { self.current_row().map_or(0, |r| r.cx2rx[self.cursor.x]) }
 
     /// Move the cursor following an arrow key (← → ↑ ↓).
-    fn move_cursor(&mut self, key: &AKey) {
+    fn move_cursor(&mut self, key: &AKey, ctrl: bool) {
+        let mut cursor_x = self.cursor.x;
         match (key, self.current_row()) {
-            (AKey::Left, Some(row)) if self.cursor.x > 0 =>
-                self.cursor.x -= row.get_char_size(row.cx2rx[self.cursor.x] - 1),
-            (AKey::Left, _) if self.cursor.y > 0 => {
-                // ← at the beginning of the line: move to the end of the previous line. The x
-                // position will be adjusted after this `match` to accommodate the current row
-                // length, so we can just set here to the maximum possible value here.
-                self.cursor.y -= 1;
-                self.cursor.x = usize::MAX;
+            (AKey::Left, Some(row)) if self.cursor.x > 0 => {
+                cursor_x -= row.get_char_size(row.cx2rx[cursor_x] - 1);
+                // ← moving to previous word
+                while ctrl && cursor_x > 0 && row.chars[cursor_x - 1] != 0x20 {
```

Comment:
I think `b' '` would be clearer than `0x20`

Reply:
I changed `0x20` to `b' '`. Thank you for feedback!